src/view/com/composer/SelectMediaBtn.tsx

```diff
@@ -0,0 +1,504 @@
+import {useCallback} from 'react'
+import {type ImagePickerAsset, launchImageLibraryAsync} from 'expo-image-picker'
+import {msg} from '@lingui/macro'
+import {useLingui} from '@lingui/react'
+
+import {
+  SUPPORTED_MIME_TYPES,
+  type SupportedMimeTypes,
+  VIDEO_MAX_DURATION_MS,
+} from '#/lib/constants'
+import {
+  usePhotoLibraryPermission,
+  useVideoLibraryPermission,
+} from '#/lib/hooks/usePermissions'
+import {getDataUriSize} from '#/lib/media/util'
+import {isNative, isWeb} from '#/platform/detection'
+import {type ComposerImage, createComposerImage} from '#/state/gallery'
+import {getVideoMetadata} from '#/view/com/composer/videos/pickVideo'
+import {atoms as a, useTheme} from '#/alf'
+import {Button} from '#/components/Button'
+import {useSheetWrapper} from '#/components/Dialog/sheet-wrapper'
+import {Image_Stroke2_Corner0_Rounded as Image} from '#/components/icons/Image'
+import {toast} from '#/components/Toast'
+
+type Props = {
+  size: number
+  disabled?: boolean
+  onAdd: (next: ComposerImage[]) => void
+  onSelectVideo: (asset: ImagePickerAsset) => void
+  setError: (error: string) => void
+}
+
+// (Note)APiligrim
+// The SelectMediaBtn is responsible for supporting the selection of the following media in the Composer:
+// up to 4 images
+// up to 1 video
+// up to 1 GIF (handled like a video - passed to onSelectVideo)
```

Comment:
i would recommend making these constants instead of a comment, which would be self-documenting